services/federator/src/Federator/Run.hs

```diff
@@ -113,17 +118,48 @@ mkCAStore settings = do
       else pure mempty
   pure (customCAStore <> systemCAStore)
 
+getClientCredentials :: RunSettings -> Either String (Maybe (FilePath, FilePath))
+getClientCredentials settings = case clientCertificate settings of
+  Nothing -> noCreds1 $> Nothing
+  Just cert -> Just . (cert,) <$> getCreds1
+  where
+    noCreds1 :: Either String ()
+    noCreds1
+      | isNothing (clientPrivateKey settings) = pure ()
+      | otherwise = Left "invalid client credentials: no certificate"
+
+    getCreds1 :: Either String FilePath
+    getCreds1 =
+      maybe (Left "invalid client credentials: no private key") pure $
+        clientPrivateKey settings
```

Comment:
Why are these called `noCreds1` and `getCreds1`?

Reply:
It's some arbitrary convention of prefixing 1 because they return (or check) credentials without the first field. Somewhat in the same spirit as `foldr1`. In this case it's just the private key, but if you pretend we had more credential fields, then it should make more sense. Hopefully :smile:. If not, I can call them something else.